AsyncDisplayKit/ASVideoPlayerNode.mm

```diff
@@ -0,0 +1,655 @@
+    if (_delegateFlags.delegatePlaybackButtonTint) {
+      _playbackButtonNode.tintColor = [_delegate videoPlayerNodePlaybackButtonTint:self];
+    } else {
+      _playbackButtonNode.tintColor = _defaultControlsColor;
+    }
+    [_playbackButtonNode addTarget:self action:@selector(playbackButtonTapped:) forControlEvents:ASControlNodeEventTouchUpInside];
+    [_cachedControls setObject:_playbackButtonNode forKey:@(ASVideoPlayerNodeControlTypePlaybackButton)];
+  }
+
+  [self addSubnode:_playbackButtonNode];
+}
+
+- (void)createElapsedTextField
+{
+  if (_elapsedTextNode == nil) {
+    _elapsedTextNode = [[ASTextNode alloc] init];
+    _elapsedTextNode.attributedString = [self timeLabelAttributedStringForString:@"00:00" forControlType:ASVideoPlayerNodeControlTypeElapsedText];
+
+    [_cachedControls setObject:_elapsedTextNode forKey:@(ASVideoPlayerNodeControlTypeElapsedText)];
+  }
+  [self addSubnode:_elapsedTextNode];
+}
+
+- (void)createDurationTextField
+{
+  if (_durationTextNode == nil) {
+    _durationTextNode = [[ASTextNode alloc] init];
+    _durationTextNode.attributedString = [self timeLabelAttributedStringForString:@"00:00" forControlType:ASVideoPlayerNodeControlTypeDurationText];
+
+    [_cachedControls setObject:_durationTextNode forKey:@(ASVideoPlayerNodeControlTypeDurationText)];
+  }
+  [self addSubnode:_durationTextNode];
+}
+
+- (void)createScrubber
+{
+  if (_scrubberNode == nil) {
+    _scrubberNode = [[ASDisplayNode alloc] initWithViewBlock:^UIView * _Nonnull{
+      UISlider *slider = [[UISlider alloc] initWithFrame:CGRectZero];
+      slider.minimumValue = 0.0;
+      slider.maximumValue = 1.0;
+
+      if (_delegateFlags.delegateScrubberMinimumTrackTintColor) {
+        slider.minimumTrackTintColor  = [_delegate videoPlayerNodeScrubberMinimumTrackTint:self];
+      }
+
+      if (_delegateFlags.delegateScrubberMaximumTrackTintColor) {
+        slider.maximumTrackTintColor  = [_delegate videoPlayerNodeScrubberMaximumTrackTint:self];
+      }
+
+      if (_delegateFlags.delegateScrubberThumbTintColor) {
+        slider.thumbTintColor  = [_delegate videoPlayerNodeScrubberThumbTint:self];
+      }
+
+      if (_delegateFlags.delegateScrubberThumbImage) {
+        UIImage *thumbImage = [_delegate videoPlayerNodeScrubberThumbImage:self];
+        [slider setThumbImage:thumbImage forState:UIControlStateNormal];
+      }
+
+
+      [slider addTarget:self action:@selector(beganSeek) forControlEvents:UIControlEventTouchDown];
+      [slider addTarget:self action:@selector(endedSeek) forControlEvents:UIControlEventTouchUpInside|UIControlEventTouchUpOutside|UIControlEventTouchCancel];
+      [slider addTarget:self action:@selector(changedSeekValue:) forControlEvents:UIControlEventValueChanged];
+
+      return slider;
+    }];
+
+    _scrubberNode.flexShrink = YES;
+
+    [_cachedControls setObject:_scrubberNode forKey:@(ASVideoPlayerNodeControlTypeScrubber)];
+  }
+
+  [self addSubnode:_scrubberNode];
```

Comment:
If you enable implicit hierarchy management, you won't need to add the subnode manually.